File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/ServerTypes/CounterpartDB/CounterpartColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CounterpartRow } from "./CounterpartRow";

export interface CounterpartColumns {
    LetterIdentifier: Column<CounterpartRow>;
    CopyType: Column<CounterpartRow>;
    CreatedDate: Column<CounterpartRow>;
}

export class CounterpartColumns extends ColumnsBase<CounterpartRow> {
    static readonly columnsKey = 'CounterpartDB.Counterpart';
    static readonly Fields = fieldsProxy<CounterpartColumns>();
}

File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/CounterpartDB/Counterpart/CounterpartColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace CorrespondenceSystem.CounterpartDB.Columns;

[ColumnsScript("CounterpartDB.Counterpart")]
[BasedOnRow(typeof(CounterpartRow), CheckNames = true)]
public class CounterpartColumns
{
    [EditLink, DisplayName("LetterIdentifier"), AlignRight]
    
    public string LetterIdentifier { get; set; }
    [Width(80)]
    public short CopyType { get; set; }
    public DateTime CreatedDate { get; set; }
    
}
   